[CI] CCSDuelIT testTerminateAfter failing

**Build Scans:**
- [elasticsearch-periodic #7377 / openjdk23_9.1.0_java-matrix-bwc](https://gradle-enterprise.elastic.co/s/jj7qzztmc7zd4)
- [elasticsearch-pull-request #65820 / 9.1.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/22w5tx4acjb7w)
- [elasticsearch-pull-request #65803 / 9.1.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/lzdgpxxjsbzj2)

**Reproduction Line:**
```
./gradlew ":qa:multi-cluster-search:v9.1.0#multi-cluster" -Dtests.class="org.elasticsearch.search.CCSDuelIT" -Dtests.method="testTerminateAfter" -Dtests.seed=EE3D1AA04CED7A7A -Dtests.bwc=true -Dtests.locale=khq-Latn-ML -Dtests.timezone=Asia/Damascus -Druntime.java=23
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.search.CCSDuelIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testTerminateAfter),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Comparing async_search minimizeRoundTrip vs. fanOut. Didn't match expected value. /terminated_early: the elements don't match: [true] != [false].
Full diff                      _clusters: 
                         details: 
                           (local): 
                             _shards: 
                                failed: same [0]
                               skipped: same [0]
                            successful: same [2]
                                 total: same [2]
                             indices: same [ccs_duel_index]
                              status: same [successful]
                           timed_out: same [false]
                                took: same [-1]
                 my_remote_cluster: 
                             _shards: 
                                failed: same [0]
                               skipped: same [0]
                            successful: same [5]
                                 total: 
[truncated]
```

**Issue Reasons:**
- [main] 3 failures in test testTerminateAfter (1.3% fail rate in 240 executions)
- [main] 2 failures in step 9.1.0_bwc-snapshots (1.7% fail rate in 117 executions)
- [main] 2 failures in pipeline elasticsearch-pull-request (2.4% fail rate in 82 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -214,9 +214,6 @@ tests:
 - class: org.elasticsearch.action.RejectionActionIT
   method: testSimulatedSearchRejectionLoad
   issue: https://github.com/elastic/elasticsearch/issues/125901
-- class: org.elasticsearch.search.CCSDuelIT
-  method: testTerminateAfter
-  issue: https://github.com/elastic/elasticsearch/issues/126085
 - class: org.elasticsearch.search.basic.SearchWithRandomDisconnectsIT
   method: testSearchWithRandomDisconnects
   issue: https://github.com/elastic/elasticsearch/issues/122707

--- a/qa/multi-cluster-search/src/test/java/org/elasticsearch/search/CCSDuelIT.java
+++ b/qa/multi-cluster-search/src/test/java/org/elasticsearch/search/CCSDuelIT.java
@@ -1333,6 +1333,7 @@ public class CCSDuelIT extends ESRestTestCase {
         Map<String, Object> responseMap = XContentHelper.convertToMap(bytesReference, false, XContentType.JSON).v2();
         assertNotNull(responseMap.put("took", -1));
         responseMap.remove("num_reduce_phases");
+        responseMap.remove("terminated_early");
         Map<String, Object> profile = (Map<String, Object>) responseMap.get("profile");
         if (profile != null) {
             List<Map<String, Object>> shards = (List<Map<String, Object>>) profile.get("shards");

